lib/postgres/models/user.go

```diff
@@ -3,12 +3,17 @@ package models
 type RoleEnum string
 
 const (
-	AdminRole RoleEnum = "admin"
-	UserRole  RoleEnum = "user"
+	AdminRole RoleEnum = "ADMIN"
+	UserRole  RoleEnum = "USER"
 )
 
 type User struct {
-	ID    string   `json:"id" gorm:"column:id;primaryKey"`
-	Role  RoleEnum `json:"role" gorm:"column:role;type:role;not null"`
-	Email string   `json:"email" gorm:"column:email;not null"`
+	Id       *string   `json:"id" gorm:"column:id;primaryKey;default:gen_random_uuid()"`
```

Comment:
Why is Id a pointer?

Reply:
oops, this was back i was trying to send it as NULL

fixed